discovergy/discovergy.py

```diff
@@ -42,14 +42,20 @@ def _fetch_consumer_tokens(self):
                                      data={'client': self._client_name},
                                      headers={},
                                      timeout=TIMEOUT)
+        except Exception as e:
+            logger.error('Failed consumer token request: %s', e)
+
+        try:
             self._oauth_key = response.json()['key']
-            self._oauth_secret = response.json()['secret']
-            return response
+        except Exception as e:
+            logger. error('Missing key in response: %s', e)
 
+        try:
+            self._oauth_secret = response.json()['secret']
         except Exception as e:
-            message = exception_template.format(type(e).__name__, e.args)
```

Comment:
Are you aware of what the exception template is doing? Did you change that on purpose and if so, why?